### ISSUE TITLE
Hard coded protocol in OdsPipeline.groovy breaks Jenkins Build

### ISSUE THREAD

**Issue by tjaeschke** [opendevstack/ods-jenkins-shared-library#74]
In line 103 of `/src/org/ods/OdsPipeline.groovy`
```
https://${context.bitbucketHost}/rest/build-status/1.0/commits/${context.gitCommit}
```
the harcoded protocol breaks the Jenkins Build, if Bitbucket is not running on https, for example in a local development environment.

**Comment by michaelsauter** [comment]
Hmm. Should we add an optional context param then, named e.g. `bitbucketScheme` to deal with this or do you have a better idea? I don't really like what I am suggesting ...

**Comment by clemensutschig** [comment]
What prevents us from adding the Scheme to the bitbucketHost variable in
the first Place ? That would fix this issue, no?

Michael Sauter <notifications@github.com> schrieb am Di., 7. Mai 2019,
23:40:

**Comment by michaelsauter** [comment]
@clemensutschig We could, but:

* All `*Host` variables should consistently include the scheme or not. Personally, I would say that a URL is made up of 4 parts: scheme, host, path and query string. So by the name of it I would guess it does NOT include the scheme...
* Ideally the change is backwards compatible, so if we add the scheme we need to do some checks in the code whether the scheme is set and default to `https://`

**Comment by gerardcl** [comment]
I vote +1 on last comment since we also had issues with Nexus Host management/usage and had to implement a fast solution to get ony the host.
So, we could indeed do that for any url...

**Comment by michaelsauter** [comment]
The proposed long-term solution is to pass URLs instead of hosts. Assigning this to ODS 3.

**Comment by michaelsauter** [comment]
See also https://github.com/opendevstack/ods-core/issues/259.

**Comment by michaelsauter** [comment]
Closing this as that specific bug is already fixed by #182.
